Geometry/HGCalCommonData/src/HGCalGeomParameters.cc

```diff
@@ -1007,23 +1026,64 @@ void HGCalGeomParameters::loadWaferHexagon8(HGCalParameters& php) {
 	++lpos;
 #ifdef EDM_ML_DEBUG
 	indtypes.emplace_back(kndx);
+	int  corners[6];
```

Comment:
It looks to me like the size of "corners" should be HGCalParameters::k_CornerSize, because that value governs the limit of the index "k". If HGCalParameters::k_CornerSize is ever greater than 6, this code will overwrite memory when compiled for debugging.

Reply:
This dimension is correct. Please do not confuse with k_CornerSize which is done in the context of providing 8 or 12 corners of calorimeter cells

Reply:
As far as I can tell, k_CornerSize is 6. It is defined in ​Geometry/​HGCalCommonData/​interface/​HGCalParameters.h:
static constexpr uint32_t k_CornerSize   = 6;

It would seem to be exactly the constant to use to define the "corners" array. This constant also controls the size of the "k" index that is used to assign to the "corners" array. Why not use this constant? What am I missing?

Reply:
No it will not corners are used differently in different scenarios. Here it deals only one plane. This will be clarified in future PR's

Reply:
I created an issue for this: https://github.com/cms-sw/cmssw/issues/25078

Reply:
Sorry - I am using the constant now

Reply:
Thanks for the fixes.